common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp

```diff
@@ -281,12 +295,51 @@ bool ap_wlan_hal_nl80211::switch_channel(int chan, int bw, int vht_center_freque
     LOG(TRACE) << __func__ << " channel: " << chan << ", bw: " << bw
                << ", vht_center_frequency: " << vht_center_frequency;
 
-    // TODO: IMPLEMENT!
+    std::string cmd = "CHAN_SWITCH 5 ";
 
-    LOG(DEBUG) << "Got channel switch, simulate ACS-STARTED;ACS-COMPLETED";
-    event_queue_push(Event::ACS_Started);
-    event_queue_push(Event::ACS_Completed);
-    event_queue_push(Event::CSA_Finished);
+    if (chan == 0) {
+        LOG(ERROR) << "ACS i not supported";
+        return false;
+    } else {
+
+        std::string bandwidth_str =
+            std::to_string(beerocks::utils::convert_bandwidth_to_int((beerocks::eWiFiBandwidth)bw));
```

Comment:
It is confusing that `bw` argument is and ineger although it is treated as an enum. On dwpal implementation the int is not representing an enum, so here should be the same.

Reply:
There is a bug in dwpal,

in channel_selection_task:
channel_switch_request.bandwidth = beerocks::BANDWIDTH_80;

in ap_manager_thread print before switch call:
beerocks::utils::convert_bandwidth_to_int(
                          (beerocks::eWiFiBandwidth)request->cs_params().bandwidth)

So bw is passed as enum.

Reply:
It seems to me like a bug in the ap_manager.
I mean, it is better to align with the integer representation rather than the enum.

Reply:
It's used as enum in all flow, starting from controller/cli/chan selection task.